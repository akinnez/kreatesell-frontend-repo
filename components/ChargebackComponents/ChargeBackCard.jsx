import Image from 'next/image';
import {GrossSales, Profit, UnitSales, Visit} from '../../utils';
import styles from './ChargeBackCard.module.scss';

export const ChargeBackCard = ({pending, won = 0, lost = 0, declined = 0}) => {
	return (
		<div className={styles.container}>
			<div className={styles.midSection}>
				<div className={styles.cardContainer}>
					<div className={styles.firstCard}>
						<div className={styles.iconCont}>
							<div className={styles.icon}>
								<Image
									src={Visit}
									width="60"
									height="60"
									alt="card icon"
								/>
							</div>
							<div className={styles.countCont}>
								<div className={styles.value}>{pending}</div>
								<div className={styles.valueText}>Pending</div>
							</div>
						</div>
					</div>

					<div className={styles.card}>
						<div className={styles.iconCont}>
							<div className={styles.icon}>
								<Image
									src={UnitSales}
									width="60"
									height="60"
									alt="card icon"
								/>
							</div>
							<div className={styles.countCont}>
								<div className={styles.value}>{won}</div>
								<div className={styles.valueText}>Won</div>
							</div>
						</div>
					</div>

					<div className={styles.card}>
						<div className={styles.iconCont}>
							<div className={styles.icon}>
								<Image
									src={GrossSales}
									width="60"
									height="60"
									alt="card icon"
								/>
							</div>
							<div className={styles.countCont}>
								<div className={styles.value}>{lost}</div>
								<div className={styles.valueText}>Lost</div>
							</div>
						</div>
					</div>

					<div className={styles.card}>
						<div className={styles.iconCont}>
							<div className={styles.icon}>
								<Image
									src={Profit}
									width="60"
									height="60"
									alt="card icon"
								/>
							</div>
							<div className={styles.countCont}>
								<div className={styles.value}>{declined}</div>
								<div className={styles.valueText}>Decline</div>
							</div>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};
