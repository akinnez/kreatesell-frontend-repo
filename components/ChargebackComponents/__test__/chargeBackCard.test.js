import {ChargeBackCard} from '../ChargeBackCard';
import renderer from 'react-test-renderer';

const mockProps = {
	pending: 'mockText',
	won: 0,
	lost: 0,
	declined: 0,
};

it('renders a snapshot of <ChargeBackCard /> with mock props', () => {
	const card = renderer.create(<ChargeBackCard {...mockProps} />).toJSON();
	expect(card).toMatchSnapshot();
});
