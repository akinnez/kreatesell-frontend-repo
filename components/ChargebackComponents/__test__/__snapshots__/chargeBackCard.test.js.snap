// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <ChargeBackCard /> with mock props 1`] = `
<div
  className="container"
>
  <div
    className="midSection"
  >
    <div
      className="cardContainer"
    >
      <div
        className="firstCard"
      >
        <div
          className="iconCont"
        >
          <div
            className="icon"
          >
            <img
              src="[jest.find('../utils')]"
            />
          </div>
          <div
            className="countCont"
          >
            <div
              className="value"
            >
              mockText
            </div>
            <div
              className="valueText"
            >
              Pending
            </div>
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="iconCont"
        >
          <div
            className="icon"
          >
            <img
              src="[jest.find('../utils')]"
            />
          </div>
          <div
            className="countCont"
          >
            <div
              className="value"
            >
              0
            </div>
            <div
              className="valueText"
            >
              Won
            </div>
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="iconCont"
        >
          <div
            className="icon"
          >
            <img
              src="[jest.find('../utils')]"
            />
          </div>
          <div
            className="countCont"
          >
            <div
              className="value"
            >
              0
            </div>
            <div
              className="valueText"
            >
              Lost
            </div>
          </div>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="iconCont"
        >
          <div
            className="icon"
          >
            <img
              src="[jest.find('../utils')]"
            />
          </div>
          <div
            className="countCont"
          >
            <div
              className="value"
            >
              0
            </div>
            <div
              className="valueText"
            >
              Decline
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
